treeherder/perfalert/perfalert/__init__.py

```diff
@@ -89,25 +89,17 @@ def __repr__(self):
                                                  self.t, self.state)
 
 
-class Analyzer:
-    def __init__(self):
-        self.data = []  # List of PerfDatum instances
-
-    def add_data(self, push_timestamp, value, **kwargs):
-        self.data.append(Datum(push_timestamp, value, **kwargs))
-        self.data.sort()
-
-    def analyze_t(self, min_back_window=12, max_back_window=24,
+def detect_changes(data=[], min_back_window=12, max_back_window=24,
                   fore_window=12, t_threshold=7):
         # Use T-Tests
```

Comment:
You need to adjust the indentation for the function body here along with the method signature. Other than that this looks good! Looking forward to seeing the rest of your changes.